server/expression/executor_test.go

```diff
@@ -291,6 +291,11 @@ func TestArrayExecution(t *testing.T) {
 			Query:      `'{ "array": [{ "name": "john", "age": 37 }, { "name": "jonas", "age": 38 }]}' | json_path '$.array[*].age' = [37, 38]`,
 			ShouldPass: true,
 		},
+		{
+			Name:       "arrays_can_be_filtered_by_value",
+			Query:      `'[{ "name": "john", "age": 37 }, { "name": "jonas", "age": 38 }]' | json_path '$[?(@.name == "john")].age' = 37`,
+			ShouldPass: true,
+		},
```

Comment:
@danielbdias asked me if the library we use for json_path supports searching an array item by one of it's fields value. Added this test to ensure that.